### PR TITLE
Add note that input data should be whitened

### PR DESCRIPTION
**Pull request by cbrnr**
A useful little note for people like me who want to use the `infomax` function directly without going through the class.

### CODE REVIEW

**Comment by cbrnr** [comment]
In addition, `ICA.fit` doesn't seem to whiten by default either (`None` just standardized the channels), so this is not just an issue with the `infomax` function. Most ICA implementations use the whitened data as a starting point to simplify the optimization, e.g. FastICA (section 5.2 in [this paper](https://www.cs.helsinki.fi/u/ahyvarin/papers/NN00new.pdf)) or the Infomax implementation in EEGLAB. The identity matrix for the initial weights also only makes sense if the data is pre-whitened.

File: mne/preprocessing/infomax_.py [hunk 1]
```diff
@@ -23,6 +23,7 @@ def infomax(data, weights=None, l_rate=None, block=None, w_change=1e-12,
     ----------
     data : np.ndarray, shape (n_samples, n_features)
         The data to unmix.
+        .. note:: The data should be whitened prior to performing Infomax ICA.
```
Comment (agramfort) [inline mne/preprocessing/infomax_.py]: does this render fine?

I would put this below in the notes section and write as description.

The whitened data to unmix.

**Comment by agramfort** [comment]
are you saying we don't properly whiten by default? Indeed whitening is required by our ICA implementations.

File: mne/preprocessing/infomax_.py (hunk 1 above)
Comment (cbrnr) [inline mne/preprocessing/infomax_.py]: There is a similar note for the `l_rate` argument - should we collect these notes and mention them in a "Notes" section before or after the references?

**Comment by cbrnr** [comment]
I'm not sure, but the default value for `noise_cov` in the `ICA` constructor is `None`, which means that the channels are just z-scored and not decorrelated/whitened. Slightly off-topic: I think the name `noise_cov` is a bit misleading, because when I compute ICA on EEG data, I guess I need to provide the EEG covariance matrix (what noise?).

**Comment by agramfort** [comment]
@cbrnr noise cov is for scaling channels eg grad, mag and eeg before doing PCA + whitening.

**Comment by agramfort** [comment]
ready to merge?

**Comment by cbrnr** [comment]
Yes, please go ahead and merge.

What about the whitening process? Regarding the noise cov, how is it estimated and how is it used for whitening? Isn't it just the cov matrix for the different signal types? I.e., if `x` is a `channels, samples` matrix of EEG channels, then the noise cov is just the covariance of `x`?

**Comment by agramfort** [comment]
noise cov can be an instance of Covariance most often used for inverse modeling. If None it scales each channel type (not each channel individually). See _pre_whiten method in ica.py

**Comment by cbrnr** [comment]
So this sounds like there is no whitening by default, but just standardization. Right?

**Comment by agramfort** [comment]
yes standardization by type of channel. You can see this as whitening with
a noise cov model which is diagonal and with one value per channel type. PR
with doc improvements welcome if you think it would help

**Comment by cbrnr** [comment]
Hm, yes, I'll think about it. I think this could be unclear for people who are not familiar with MNE, but want to use MNE's ICA implementation. It's a good start that this is now mentioned in the `infomax` function.

**Comment by agramfort** [comment]
if you have 1 channel type then noise_cov=None is doing the standard
pipeline

**Comment by cbrnr** [comment]
But then this is not proper whitening. For example, if I only have EEG, standardizing the channels does not decorrelate/whiten them.

**Comment by agramfort** [comment]
yes but the whitening is done later in the _fit method via PCA(whiten=True,
...)

**Comment by cbrnr** [comment]
Ah, I see - then everything is fine :-)

**Pull request merged**